client/checkout/utils/upe.js

```diff
@@ -331,26 +337,106 @@ export const togglePaymentMethodForCountry = ( upeElement ) => {
+	}
 
 	/* global wcpayCustomerData */
 	// in the case of "pay for order", there is no "billing country" input, so we need to rely on backend data.
 	const billingCountry =
-		document.getElementById( 'billing_country' )?.value ||
-		wcpayCustomerData?.billing_country ||
-		'';
+		billingInput?.value || wcpayCustomerData?.billing_country || '';
 
-	const upeContainer = document.querySelector(
-		'.payment_method_woocommerce_payments_' + paymentMethodType
-	);
+	const upeContainer = upeElement?.closest( '.wc_payment_method' );
 	if ( supportedCountries.includes( billingCountry ) ) {
 		upeContainer.style.removeProperty( 'display' );
 	} else {
 		upeContainer.style.display = 'none';
-		// if the toggled off payment method was selected, we need to fall back to credit card
 		if ( paymentMethodType === selectedPaymentMethod ) {
-			document
-				.querySelector( '#payment_method_woocommerce_payments' )
-				.click();
+			const defaultPaymentMethod = 'card';
+			const cardPaymentForm = document.querySelector(
+				`.wcpay-upe-form[data-payment-method-type="${ defaultPaymentMethod }"]`
+			);
+
+			const cardPaymentMethodInput = cardPaymentForm
+				?.closest( '.wc_payment_method' )
+				.querySelector(
+					`input[name="payment_method"][value="${ cardPaymentForm.dataset.gatewayId }"]`
+				);
+
+			cardPaymentMethodInput?.click();
 		}
 	}
 };
+
+function getParsedLocale() {
+	try {
+		return JSON.parse(
+			wc_address_i18n_params.locale.replace( /&quot;/g, '"' )
+		);
+	} catch ( e ) {
+		return null;
+	}
+}
+
+export const isBillingInformationMissing = ( $form ) => {
+	const form = $form[ 0 ];
+	const enabledBillingFields = getUPEConfig( 'enabledBillingFields' );
+
+	// first name and last name are kinda special - we just need one of them to be at checkout
+	const name = `${
+		form.querySelector(
+			`#${ SHORTCODE_BILLING_ADDRESS_FIELDS.first_name }`
+		)?.value || ''
+	} ${
+		form.querySelector( `#${ SHORTCODE_BILLING_ADDRESS_FIELDS.last_name }` )
+			?.value || ''
+	}`.trim();
+	if (
+		! name &&
+		( enabledBillingFields[ SHORTCODE_BILLING_ADDRESS_FIELDS.first_name ] ||
+			enabledBillingFields[ SHORTCODE_BILLING_ADDRESS_FIELDS.last_name ] )
+	) {
+		return true;
+	}
+
+	const billingFieldsToValidate = [
+		'billing_email',
+		SHORTCODE_BILLING_ADDRESS_FIELDS.country,
+		SHORTCODE_BILLING_ADDRESS_FIELDS.address_1,
+		SHORTCODE_BILLING_ADDRESS_FIELDS.city,
+		SHORTCODE_BILLING_ADDRESS_FIELDS.postcode,
+	].filter( ( field ) => enabledBillingFields[ field ] );
+
+	const country = billingFieldsToValidate.includes(
+		SHORTCODE_BILLING_ADDRESS_FIELDS.country
+	)
+		? form.querySelector( `#${ SHORTCODE_BILLING_ADDRESS_FIELDS.country }` )
+				?.value
+		: null;
+
+	// We need to just find one field with missing information. If even only one is missing, just return early.
+	return Boolean(
+		billingFieldsToValidate.find( ( fieldName ) => {
+			const $field = form.querySelector( `#${ fieldName }` );
```

Comment:
I think this was pre-existing, but the `$` prefix is typically used for jQuery elements
```suggestion
			const field = form.querySelector( `#${ fieldName }` );
```